pkg/rebuild/rebuild/rebuildremote.go

```diff
@@ -110,40 +127,68 @@ var assetUploadTpl = template.Must(
 	))
 
 func makeBuild(t Target, dockerfile string, opts RemoteOptions) (*cloudbuild.Build, error) {
+	var cleanupScript bytes.Buffer
+	err := rebuildCleanupTpl.Execute(&cleanupScript, map[string]any{
+		"Artifact": transfer{
+			From: path.Join("/out", t.Artifact),
+			To:   path.Join("/workspace", t.Artifact),
+		},
+		"EnableEBPF": opts.EnableEBPF,
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "expanding asset upload template")
+	}
 	var assetUploadScript bytes.Buffer
-	err := assetUploadTpl.Execute(&assetUploadScript, map[string]any{
+	var uploads = []transfer{
+		{From: "/workspace/image.tgz", To: opts.RemoteMetadataStore.URL(Asset{Target: t, Type: ContainerImageAsset}).String()},
+		{From: path.Join("/workspace", t.Artifact), To: opts.RemoteMetadataStore.URL(Asset{Target: t, Type: RebuildAsset}).String()},
+	}
+	if opts.EnableEBPF {
+		uploads = append(uploads, transfer{From: "/workspace/tetragon.log", To: opts.RemoteMetadataStore.URL(Asset{Target: t, Type: TetragonLog}).String()})
+	}
+	err = assetUploadTpl.Execute(&assetUploadScript, map[string]any{
 		"UtilPrebuildBucket": opts.UtilPrebuildBucket,
-		"Uploads": []upload{
-			{From: "/workspace/image.tgz", To: opts.RemoteMetadataStore.URL(Asset{Target: t, Type: ContainerImageAsset}).String()},
-			{From: path.Join("/workspace", t.Artifact), To: opts.RemoteMetadataStore.URL(Asset{Target: t, Type: RebuildAsset}).String()},
-		},
+		"Uploads":            uploads,
 	})
 	if err != nil {
 		return nil, errors.Wrap(err, "expanding asset upload template")
 	}
-	return &cloudbuild.Build{
+	var build = cloudbuild.Build{
 		LogsBucket:     opts.LogsBucket,
 		Options:        &cloudbuild.BuildOptions{Logging: "GCS_ONLY"},
 		ServiceAccount: opts.BuildServiceAccount,
 		Steps: []*cloudbuild.BuildStep{
 			{
 				Name:   "gcr.io/cloud-builders/docker",
 				Script: "set -eux\ncat <<'EOS' | docker buildx build --tag=img -\n" + dockerfile + "\nEOS\ndocker run --name=container img",
+				Id:     "run_builder",
+				// Run immediately ("-" is the id of the start of the build).
+				WaitFor: []string{"-"},
 			},
 			{
-				Name: "gcr.io/cloud-builders/docker",
-				Args: []string{"cp", "container:" + path.Join("/out", t.Artifact), path.Join("/workspace", t.Artifact)},
+				Name:    "gcr.io/cloud-builders/docker",
+				Script:  cleanupScript.String(),
+				Id:      "cleanup",
+				WaitFor: []string{"run_builder"},
 			},
 			{
-				Name:   "gcr.io/cloud-builders/docker",
-				Script: "docker save img | gzip > /workspace/image.tgz",
-			},
-			{
-				Name:   "gcr.io/cloud-builders/gsutil",
-				Script: assetUploadScript.String(),
+				Name:    "gcr.io/cloud-builders/gsutil",
+				Script:  assetUploadScript.String(),
+				WaitFor: []string{"cleanup"},
 			},
 		},
-	}, nil
+	}
+	if opts.EnableEBPF {
+		// Prepend a build step to start the tetragon contianer.
+		build.Steps = append([]*cloudbuild.BuildStep{{
```

Comment:
is there a way to avoid the after-the-fact prepend so we can make these steps more sequentially defined?

Reply:
I constructed steps separate above, how's that?